src/load_inputs/load_generators_data.jl

```diff
@@ -501,4 +496,47 @@ function load_vre_stor_data!(inputs_gen::Dict, setup::Dict, path::AbstractString
 		inputs_gen["dfVRE_STOR"] = DataFrame()
 	end
 	summarize_errors(error_strings)
-end
\ No newline at end of file
+end
+
+
+function process_piecewisefuelusage!(inputs::Dict, scale_factor)
+	gen_in = inputs["dfGen"]
+	inputs["PWFU_Max_Num_Segments"] = 0
+	inputs["THERM_COMMIT_PWFU"] = Int64[]
+
+	if any(occursin.(Ref("PWFU_"), names(gen_in)))
+		slope_mat = extract_matrix_from_dataframe(gen_in, "PWFU_Slope")
+		intercept_mat = extract_matrix_from_dataframe(gen_in, "PWFU_Intercept")
+		if size(slope_mat)[2] != size(intercept_mat)[2]
+			@error """ The number of slope and intercept columns used for piecewise fuel consumption must be equal
+			"""
+		end
+
+        # check if values for piecewise fuel consumption make sense. Negative slopes are not allowed as they will make the model non-convex
+        if any(slope_mat .< 0)
+			@error """ Slope used for piecewise fuel consumption cannot be negative
+			"""
+		end
+
+		# identify nonzero slope and intercept for each generator, and the "or" matrix will return a set of generators that contain slope/intercept for at least one segment 
+		slope_check_zero = slope_mat .!= 0 
+        intercept_check_zero = intercept_mat .!=0
+		slope_or_intercept = slope_check_zero .| intercept_check_zero
+
+        # determine if a generator contains piecewise fuel usage segment
+		gen_in.PWFU_NUM_SEGMENTS .= any(slope_or_intercept .!=0, dims = 2)
```

Comment:
This column `gen_in.PWFU_NUM_SEGMENTS` is basically only going to be a binary value. It won't tell you how many segments each generator has. I would very much suggest renaming this, since NUM_SEGMENTS sounds like an integer. Maybe to HAS_PWFU ?